x-pack/plugins/secops/public/components/timeline/body/stateful_body.tsx

```diff
@@ -0,0 +1,230 @@
+  range?: string;
+}
+
+interface DispatchProps {
+  addNoteToEvent?: ActionCreator<{ id: string; noteId: string; eventId: string }>;
+  applyDeltaToColumnWidth?: ActionCreator<{
+    id: string;
+    columnId: string;
+    delta: number;
+  }>;
+  pinEvent?: ActionCreator<{
+    id: string;
+    eventId: string;
+  }>;
+  removeColumn?: ActionCreator<{
+    id: string;
+    columnId: string;
+  }>;
+  unPinEvent?: ActionCreator<{
+    id: string;
+    eventId: string;
+  }>;
+  updateColumns?: ActionCreator<{
+    id: string;
+    columns: ColumnHeader[];
+  }>;
+  updateRange?: ActionCreator<{
+    id: string;
+    range: string;
+  }>;
+  updateSort?: ActionCreator<{
+    id: string;
+    sort: Sort;
+  }>;
+  updateNote?: ActionCreator<{ note: Note }>;
+}
+
+type StatefulBodyComponentProps = OwnProps & ReduxProps & DispatchProps;
+
+class StatefulBodyComponent extends React.PureComponent<StatefulBodyComponentProps> {
+  public componentDidUpdate(prevProps: StatefulBodyComponentProps) {
+    const { columnHeaders, browserFields } = this.props;
+    const newHeaders = getColumnHeaders(
+      isEmpty(columnHeaders) ? defaultHeaders : columnHeaders,
+      browserFields
+    );
+    if (
+      isEmpty(columnHeaders) ||
+      (!isEmpty(prevProps.columnHeaders) && !isEqual(prevProps.columnHeaders, newHeaders))
+    ) {
+      this.onUpdateColumns(newHeaders);
+    }
+  }
+
+  public render() {
+    const {
+      browserFields,
+      columnHeaders,
+      data,
+      eventIdToNoteIds,
+      getNotesByIds,
+      height,
+      id,
+      isLoading,
+      pinnedEventIds,
+      range,
+      sort,
+      width,
+    } = this.props;
+
+    return (
+      <Body
+        addNoteToEvent={this.onAddNoteToEvent}
+        browserFields={browserFields}
+        id={id}
+        isLoading={isLoading}
+        columnHeaders={columnHeaders || []}
+        columnRenderers={columnRenderers}
+        data={data}
+        eventIdToNoteIds={eventIdToNoteIds!}
+        getNotesByIds={getNotesByIds}
+        height={height}
+        onColumnResized={this.onColumnResized}
+        onColumnRemoved={this.onColumnRemoved}
+        onColumnSorted={this.onColumnSorted}
+        onFilterChange={noop} // TODO: this is the callback for column filters, which is out scope for this phase of delivery
+        onPinEvent={this.onPinEvent}
+        onRangeSelected={this.onRangeSelected}
+        onUnPinEvent={this.onUnPinEvent}
+        pinnedEventIds={pinnedEventIds!}
+        range={range!}
+        rowRenderers={rowRenderers}
+        sort={sort}
+        updateNote={this.onUpdateNote}
+        width={width}
+      />
+    );
+  }
+
+  private onAddNoteToEvent: AddNoteToEvent = ({
+    eventId,
+    noteId,
+  }: {
+    eventId: string;
+    noteId: string;
+  }) => this.props.addNoteToEvent!({ id: this.props.id, eventId, noteId });
+
+  private onColumnSorted: OnColumnSorted = sorted => {
+    this.props.updateSort!({ id: this.props.id, sort: sorted });
```

Comment:
Looks like this is undefined -- I'm currently seeing an app crash any time I try to sort columns in the timeline after dragging over just one filter. Verified this isn't in the latest `feature-secops` branch. 
![timeline_sort_crash](https://user-images.githubusercontent.com/2946766/54244875-372e4900-44f4-11e9-8490-fe6b33a14c1c.gif)

Reply:
Good catch!!!